flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphGenerator.java

```diff
@@ -586,7 +586,7 @@ private StreamGraph generateInternal(List<StreamTransformation<?>> transformatio
 				transform.getOutputType(),
 				transform.getName());
 
-		if (transform.getStateKeySelector1() != null) {
+		if (transform.getStateKeySelector1() != null || transform.getStateKeySelector2() != null) {
```

Comment:
Do we still need the check here or will the code work if we set either to `null`? I think it does because in the broadcast case we set the second key selector to `null`.